### DISCUSSION TITLE
More advanced filtering system (LHS brackets | RHS colon) in FastAPI

### DISCUSSION THREAD

**Discussion by kostsal** [fastapi/fastapi#3396] [Questions]
Is it possible to create a filtering system in FastAPI like:

`?f=<field_name>:<operator>:<value>&f=<field_name N>:<operator>:<value 1>,<value N>`
or
`?<field_name>[<operator>]=<value>&<field_name N>[<operator>]=<value 1>,<value N>`
or
`?<field_name>=<operator>:<value>&<field_name N>=<operator>:<value 1>,<value N>`

And if we imagine that there will be a lot of such endpoints (routers and routes) and even more fields for filtering :fire: 

e.g.: lte, lt, gte, gt, in, nin, eq, ne, re (for `<operator>`)

I would like to create a unique Dependency that will work in all cases, but no ideas yet

**Comment by hamzabouissi** [comment]
I would to contribute to this project 
if u wanna let's us discuss

**Reply by andre-cavalheiro** [reply]
Did you got down to doing something around this?

**Comment by ghandic** [comment]
In OpenAPI, it supports deepObject query parameter encoding which I think you are referring to. I have an example implementation here https://github.com/ghandic/FastAPI-deepObject

However, you can always imp[element it in the body of the request and have the pydantic validation too. This is more widely used and supported as the URL length has a limit
